pdns/reczones.cc

```diff
@@ -459,19 +468,27 @@ std::shared_ptr<SyncRes::domainmap_t> parseAuthAndForwards()
       instructions = splitField(instructions, '#').first; // Remove EOL comments
       boost::trim(domain);
       boost::trim(instructions);
-      if (domain.empty() && instructions.empty()) { // empty line
-        continue;
-      }
-      if (boost::starts_with(domain, "+")) {
-        domain = domain.c_str() + 1;
-        ad.d_rdForward = true;
-      }
-      else
-        ad.d_rdForward = false;
       if (domain.empty()) {
+        if (instructions.empty()) { // empty line
+          continue;
+        }
         throw PDNSException("Error parsing line " + std::to_string(linenum) + " of " + ::arg()["forward-zones-file"]);
       }
 
+      bool allowNotifyFor = false;
+
+      for (;; domain.erase(0, 1)) {
+        switch (domain[0]) {
```

Comment:
Is there a `domain[0]` after the erase? Often at() is preferred since it has bounds-checking.
Additionally, I can see various malformed lines go undetected:
```
+^
+^=
+^;=
```
likely there are more cases. The old code is more strict.

Reply:
Good point.... I'll re-do this part.

Reply:
Next update will have this fixed, although as best I can tell the old code would have accepted `+;=`, at least through the basic parsing stage.